### PR TITLE
Add ICN header to swagger docs

### PR DESCRIPTION
**Pull request by michelpmcdonald**
## Summary

Appeals create\validate now accepts ICN as an optional header, this is prelim work for eventually replacing SSN with ICN

This PR adds the optional ICN header to our swagger docs\ and dev portal

Team Banana Peels dev, we own this.

Note, during development of this ticket a small bug in the rake task used to actually generate the swagger JSON files was discovered.  
The rake task fix is in another ticket\PR: [API-21523](https://github.com/department-of-veterans-affairs/vets-api/pull/11534)
When ^that ticket is merged, merge master into this branch, then run "rswag:appeals_api:all" to actually generate the swagger json file changes.  Commit the json file changes before merging.

There is a [release note PR](https://github.com/department-of-veterans-affairs/developer-portal-content/pull/44) for this ticket, once this is merged, update the MD file name to the date this PR will be released
and merge that ticket also.

## Related issue(s)
[API-22118](https://vajira.max.gov/browse/API-22118)

## Testing done

Manual confirmation of expected swagger json files was confirmed

## What areas of the site does it impact?
This ticket generates new swagger JSON files that the document appeals portal consumes and displays

## Acceptance criteria

- [ ]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  Documentation has been updated (link to documentation)

### CODE REVIEW

**Review by caseywilliams — CHANGES_REQUESTED** [review]
It looks like you haven't yet run the rake tasks to bring these headers into the JSON docs. You can do that by:

1. Rebasing this branch against master just in case, and
2. `bundle exec rake rswag:appeals_api:all` (Note just in case that if you haven't run migrations, jobs in this group may fail, and the output is long so it can be hard to see to see the problem if that happens)

I tried checking out your branch and doing this locally, and it looks like the JSON files should all be affected.

File: modules/appeals_api/spec/support/swagger_shared_components.rb
Comment (cilestin) [inline modules/appeals_api/spec/support/swagger_shared_components.rb:43]: A reminder that we'll want to swap this to `true` when we release the Segmented APIs, but we'll need to do so in such a way that it doesn't affect decision_reviews' endpoint docs (even if that's manually by not committing the change to their swagger*.json files).

**Review by cilestin — APPROVED** [review]

File: modules/appeals_api/spec/support/swagger_shared_components.rb
Comment (caseywilliams) [reply modules/appeals_api/spec/support/swagger_shared_components.rb:43]: I think we can use `DocHelpers.decision_reviews?` to make the false/true decision programmatically

**Review by caseywilliams — APPROVED** [review]
LGTM

**Comment by cilestin** [comment]
Noting that this will stay on hold until we get approval for the release notes on this feature

**Review by kristen-brown — APPROVED** [review]

**Pull request merged**
